src/IO/ReadHelpers.h

```diff
@@ -1106,16 +1113,33 @@ inline void readBinary(Decimal64 & x, ReadBuffer & buf) { readPODBinary(x, buf);
 inline void readBinary(Decimal128 & x, ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(Decimal256 & x, ReadBuffer & buf) { readPODBinary(x.value, buf); }
 inline void readBinary(LocalDate & x, ReadBuffer & buf) { readPODBinary(x, buf); }
-inline void readBinary(UUID & x, ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(IPv4 & x, ReadBuffer & buf) { readPODBinary(x, buf); }
 inline void readBinary(IPv6 & x, ReadBuffer & buf) { readPODBinary(x, buf); }
 
+inline void readBinary(UUID & x, ReadBuffer & buf)
+{
+    readUUIDBinary(x, buf);
+}
+
+template <typename A, typename B>
```

Comment:
Are l. 1125ff needed or leftovers?

Reply:
There was a `readBinary` needed for some types from `AggregateFunctionQuantile` that required something that looked like 

``` cpp
template<typename T>
inline void readBinary(std::pair<T, unsigned int> & x, ReadBuffer & buf);
```

I figured I'd make this more generic and keep it around.

I am realizing now though that the implementation should probably actually utilize `readBinary` instead of `readPODBinary` here. Ex: 

``` cpp
inline void readBinary(std::pair<A, B> & x, ReadBuffer & buf)
{
    readBinary(x.first, buf);
    readBinary(x.second, buf);
}
```

I'll correct this quickly.